testcases/OpTestEEH.py

```diff
@@ -379,3 +204,170 @@ def check_eeh_pe_recovery(self, pe):
+        self.cv_HOST.host_run_command(cmd)
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_IPMI.ipmi_host_login(l_con)
+        self.cv_IPMI.ipmi_host_set_unique_prompt()
+        l_con.run_command("stty cols 300")
+        l_con.run_command("stty rows 10")
+        l_con.run_command("dmesg -D")
+        l_con.run_command("uname -a")
+        l_con.run_command("cat /etc/os-release")
+        for i in range(0,2):
+            for domain in pci_domains:
+                self.prepare_logs()
+                cmd = "echo 0x8000000000000000 > /sys/kernel/debug/powerpc/%s/err_injct_outbound; lspci;" % domain
+                print "=================Injecting the fenced PHB error on PHB: %s=================" % domain
+                l_con.run_command(cmd)
+                # Give some time to EEH PCI Error recovery
+                time.sleep(30)
+                l_con.sendline("\r\n")
+                l_con.sendline("\r\n")
+                self.gather_logs()
+                if i == 0:
+                    if not self.check_eeh_phb_recovery(domain):
+                        msg = "PHB domain %s recovery failed" % domain
+                        raise OpTestError(msg)
+                    else:
+                        print "PHB %s recovery successful" % domain
+                else:
+                    if self.check_eeh_phb_recovery(domain):
+                        msg = "PHB domain %s not removed from the system" % domain
+                        raise OpTestError(msg)
+                    else:
+                        print "PHB domain %s removed successfully" % domain
+
+class OpTestEEHbasic_frozen_pe(OpTestEEH):
+    ##
+    # @brief  This testcase has below steps
+    #         1. Get the list of pci PHB domains
+    #         2. Get the root PHB domain where the root file system
+    #            is installed(We need to skip this as EEH recovery will
+    #            fail on root PHB).
+    #         3. get dictionary of pe vs config addr
+    #            Ex: {'0001:0c:00.2': '2', '0001:0b:00.0': 'fb', '0001:0c:00.0': '2'}
+    #         4.Prepare below command & Start inject frozenPE errors on all PE's
+    #           echo "PE_number:<0,1>:<function>:0:0" > /sys/kernel/debug/powerpc/PCIxxxx/err_injct
+    #         5. Gather necssary logs(dmesg & OPAL) and check the device(PE) got
+    #            recovered or not.
+    #
+    # @return BMC_CONST.FW_SUCCESS or BMC_CONST.FW_FAILED
+    #
+    def runTest(self):
+        root_domain = self.cv_HOST.host_get_root_phb()
+        pci_domains = self.cv_HOST.host_get_list_of_pci_domains()
+        print "Skipping the root phb %s for frozen PE Testcase" % root_domain
+        pci_domains.remove(root_domain)
+        pe_dic = self.get_dic_of_pe_vs_addr()
+        self.cv_HOST.host_gather_opal_msg_log()
+        self.cv_HOST.host_gather_kernel_log()
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_IPMI.ipmi_host_login(l_con)
```

Comment:
Here also same.
console = self.cv_SYSTEM.sys_get_ipmi_console() # IPMIConsole object not the sol session object
con = console.get_console() # Get the actual IPMI SOL Console
self.cv_IPMI.ipmi_host_login(con) # Login to the host on IPMI console
self.cv_IPMI.ipmi_host_set_unique_prompt() # Set unique prompt
console.run_command(cmd, timeout) # Run the actual commands on host console

Better to use good naming and convention so that users won't confuse b/w solchild and IPMIConsole class objects